featurebyte/service/online_serving.py

```diff
@@ -195,9 +199,21 @@ async def get_online_features_by_feast(
 
         feast_online_features = feast_store.get_online_features(feature_service, request_data)
 
-        return OnlineFeaturesResponseModel(
-            features=feast_online_features.to_df().to_dict(orient="records")
+        # Map feature names to the original names
+        feature_docs = await self.feature_service.list_documents_as_dict(
+            query_filter={"_id": {"$in": feature_list.feature_ids}},
+            projection={"name": 1, "version": 1},
         )
+        feature_name_map = {}
+        for feature_doc in feature_docs["data"]:
+            feature_name = feature_doc["name"]
+            feature_version = VersionIdentifier(**feature_doc["version"]).to_str()
+            feature_name_map[f"{feature_name}_{feature_version}"] = feature_name
+
+        features = (
+            feast_online_features.to_df().rename(columns=feature_name_map).to_dict(orient="records")
+        )
+        return OnlineFeaturesResponseModel(features=features)
```

Comment:
Map the features back to original name (without version) before returning online features response.